packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1237,39 +1237,42 @@ bool hasPlugins(FlutterProject project) {
 ///
 ///  For more details, https://flutter.dev/go/federated-plugins.
 // TODO(stuartmorgan): Expand implementation to apply to all implementations,
-// not just Dart-only, per the federated plugin spec.
+//  not just Dart-only, per the federated plugin spec.
+// TODO(gustl22): The flag [throwOnPluginPubspecError] is currently only used
+//  for testing dart plugins as the logic is not working correctly.
 List<PluginInterfaceResolution> resolvePlatformImplementation(
   List<Plugin> plugins, {
   bool throwOnPluginPubspecError = true,
 }) {
-  final List<String> platforms = <String>[
+  const Iterable<String> platformKeys = <String>[
```

Comment:
out of curiosity, why change the type here to `Iterable`?

Reply:
I like to follow the habit of using the most abstract type which still fulfills the functionality: See Facade and Template method [design patterns](https://en.wikipedia.org/wiki/Design_Patterns), which allows to swap the underlying logic (e.g. to be able to replace it with a Set)

I followed this pattern also [here](https://github.com/flutter/flutter/pull/141780/files#diff-68361f6fc9aafa5fa7de042eec4603dd4f321eb783a00edd57780ef78c1094bbR169)

But I also have no pain keeping it as a List.

Reply:
no, this is fine